app/src/main/java/org/jellyfin/mobile/player/PlayerViewModel.kt

```diff
@@ -69,9 +71,14 @@ class PlayerViewModel(application: Application) : AndroidViewModel(application),
         viewModelScope.launch {
             for (event in playerEventChannel) {
                 when (event) {
-                    PlayerEvent.PAUSE -> mediaSessionCallback.onPause()
-                    PlayerEvent.RESUME -> mediaSessionCallback.onPlay()
-                    PlayerEvent.STOP, PlayerEvent.DESTROY -> mediaSessionCallback.onStop()
+                    PlayerEvent.Pause -> mediaSessionCallback.onPause()
+                    PlayerEvent.Resume -> mediaSessionCallback.onPlay()
+                    PlayerEvent.Stop, PlayerEvent.Destroy -> mediaSessionCallback.onStop()
+                    is PlayerEvent.Seek -> playerOrNull?.seekTo(event.ms)
+                    is PlayerEvent.SetVolume -> {
+                        setVolume(event.volume)
+                        reportPlaybackState()
```

Comment:
> Why do we need to report the state for volume changes?

Because the volume than "jumped" on the webclient. This fixes it, because it instantly reports back.

> And shouldn't we do it for other events (like pause, seek etc.) too?

That's already happening, when you seek/pause/etc. the player state changes, triggering the ExoPlayer callback in the ViewModel. For volume, it just needs an extra invocation because that isn't handled by ExoPlayer. Honestly, I might even have to add a volume listener (if that exists) to also update if you change with the volume buttons, and not have the timedupdates delay first.